### ISSUE TITLE
Subscriber rendered outside of context of a Broadcast but inside of another Broadcast for a different concern

### ISSUE THREAD

**Issue by landabaso** [ReactTraining/react-broadcast#29]
@mjackson, @ryanflorence thanks for the lib!

I'm getting an error because in occasions I wrap certain components with `<Subscriber>` when they are not descendants of  `<Broadcast>`. I know this was discussed (https://github.com/ReactTraining/react-broadcast/issues/6) and fixed but I'm still getting this error:

`Uncaught Error: <Subscriber channel="concernX"> must be rendered in the context of a <Broadcast channel="concernX">`

I should mention that I use different `<Broadcast> - <Subscriber>` pairs for different concerns.

For example:

```
<Broadcast channel="concern1">
  <Broadcast channel="concern2">
    {whatever}
  </Broadcast>
  <Subscriber channel="concern2">
    {whateverElse}
  </Subscriber>
</Broadcast>
```

There's a problem in `<Subscriber channel="concern2">` because `this.context.broadcasts` exists but not for `concern2`.

Is this the expected behaviour? Any workaround?

**Comment by mjackson** [comment]
Yes, this is expected behavior. We can't just let a `<Subscriber>` fail silently in that scenario. There's nothing to subscribe to!

**Comment by landabaso** [comment]
Thanks for clarification @mjackson.

I read there were plans to allow using `<Subscriber>` out of context after reading https://github.com/ReactTraining/react-broadcast/issues/6 and got confused.

[There was a commit](https://github.com/ReactTraining/react-broadcast/commit/57de4ff3cd9a157eebc9756d4ed6a794b7e7eada) that allowed that (but was not working for my particular case because of the reasons explained). I checked the code again and I realised there were further commits that changed that.
